text/trace/0170-sampling-probability.md

```diff
@@ -580,92 +586,268 @@ The OTEP 168 proposal for propagation limits head sampling
+of composition we assign unknowns `p=64`, which is 1 beyond the range
+of the 6-bit that represent known p-values.  The assignment of `p=64`
+simplifies the formulas below .
+
+By folowing these simple rules, any numher of consistent probability
+samplers and non-probability samplers can be combined.  Starting with
+`p=64` representing unknown and `sampled=false`, update the composite
+p-value to the minimum value of the prior composite p-value and the
+individual sampler p-value.
 
 ```
-  // Log-head-adjusted count is the logarithm of adjusted count for
-  // this span as calculated at the head, offset by +1, with the
-  // following recognized values.
-  //
-  // 0: The zero value represents an UNKNOWN adjusted count.
-  //    Consumers of these Spans cannot cannot compute span metrics.
-  //
-  // 1: An adjusted count of 1.
-  //
-  // 2-62: Values 2 through 62 represent an adjusted count of 2^(Value-1)
-  //
-  // 63: Value 63 represents an adjusted count of zero.
-  //
-  // Values greater than 64 are unrecognized.
-  uint32 log_head_adjusted_count = <next_tag>;
+p<sub>out</sub> = min(p<sub>in</sub>, p<sub>sampler</sub)
+sampled<sub>out</sub> = logicalOR(sampled<sub>in</sub>, sampled<sub>sampler</sub)
 ```
 
+The composite sampler is always the logical-OR of the individual
+samplers.  For p-value, this has two effects:
+
+1. When combining two consistent probability samplers, the
+less-selective Sampler's adjusted count is taken.
+2. When combining a consistent probability sampler and a
+non-probability sampler, this has the effect of changing unknown
+adjusted count into known adjusted count.
+
 ### Proposed `Sampler` interface changes
 
 The Trace SDK specification of the `SamplingResult` will be extended
 with a new field to be returned by all Samplers.
 
 ```
-- The sampling probability of the span is encoded as one plus the
-  inverse of head inclusion probability, known as "adjusted count",
-  which is the effective count of the Span for use in Span-to-Metrics
-  pipelines.  The value 0 is used to represent unknown adjusted count,
-  and the value 63 is used to represent known-zero adjusted count.
-  For values >0 and <63, the adjusted count of the Span is
-  2^(value-1), representing power-of-two probabilities between
-  1 and 2^-61.
+- The sampling probability of the span is encoded as the base-2
+  logarithm of inverse head inclusion probability, known as "adjusted
+  count", which is the effective count of the Span for use in
+  Span-to-Metrics pipelines.  The value 64 is used to represent
+  unknown adjusted count, and the value 63 is used to represent
+  known-zero adjusted count.  For values >=0 and <63, the adjusted
+  count of the Span is 2**value, representing power-of-two
+  probabilities between 1 and 2**-62.
 
   The corresonding `SamplerResult` field SHOULD be named
-  `log_head_adjusted_count` to match the Span data model.
+  `log_head_adjusted_count` because it is the logarithm of the head
```

Comment:
for the same reasons as for head sampling probability, I would prefer `log_adjusted_count` over `log_head_adjusted_count`

Reply:
Done.